### ISSUE TITLE
Request to add more movie/tv show databases

### ISSUE THREAD

**Issue by Omarscomin** [Tikipeter/repository.tikipeter#283]
Hello Tikipeter, I hope you're doing well! Thank you for all of the hard work that you've put into Fen. 

I was wondering if it might be possible to add more movie/tv show databases, such as IMDB and Trakt, to the current movie/tv show browsing sections (trending, popular, recently released, genres, etc). The reason I ask is because TMDB tends to suggest content that is foreign to the country/language that I have set for both Kodi & Fen. I recently obtained my own API key from TMDB, hoping it would help, but it unfortunately made no difference. 

Definitely no big deal if you aren't interested in doing this. Thank you again for everything! I hope you have a great one!😊

**Comment by Tikipeter** [comment]
I only get English speaking movies and tv shows.

What country/language do you have set?

What lists are giving foreign titles?

Are you sure you've gone to the correct github? I know there's another addon that you use.... :-)

**Comment by Omarscomin** [comment]
> I only get English speaking movies and tv shows.
> 
> 
> 
> What country/language do you have set?
> 
> 
> 
> What lists are giving foreign titles?
> 
> 
> 
> Are you sure you've gone to the correct github? I know there's another addon that you use.... :-)

I have English set as well. A Polish movie appears on the first page of "Popular" movies ("Mr. Car and the Knights Templar", 2023) and a Spanish movie appears on the second page ("My Fault", 2023). Neither film is particularly popular either, which is a little annoying when there's only 20 titles per page. The fact that IMDB lists and Trakt lists display more titles per page than TMDB lists do is actually another reason I prefer them. 

Don't get smart with me, Mr, Big Balls Fenny. I'm definitely not lost. I even tried enabling the "make Fen work properly" setting before coming here ;) lol!

**Comment by Tikipeter** [comment]
I get neither of those 2 titles in the first 2 pages of Popular. I do, however, get Mr. Car on page one and My Fault on page two of a very popular addon that is in some ways similar to Fen. You wouldn't have screenshots would you?...

Anyway....

TRAKT: "Trending", "Trending Recent", "Top 10 Box Office" and "Most Watched" are all currently being offered in Fen. There really is not much more that trakt offers in the way of useful, pre-compiled lists to use.

IMDb: I currently offer 0 IMDb based pre-compiled lists. The main reason for this is that it takes an extra call to TMDb to get the metadata for each movie/show for these lists. Instead of 1 call, it's 2 as you can only get the IMDb ID from the lists, and TMDb requires a TMDb ID for accessing the metadata. So you need to first make a call to TMDb to get the TMDb ID from the IMDb ID. Plus IMDb doesn't offer a public API, so the lists are scraped from their website which is a lot slower. Since Fen is all about speed, this has always been an issue for me.

Also, Umbrella dev and myself have different tastes in lists. He has no problem with 2 lists both being called "Popular" for example, whilst I would not do that with Fen. There are likely some nice IMDb lists that would be useful to have though. A quick look shows "Most Voted", "Featured", "Most Popular" (with a rename).

I just wouldn't do 2 or more lists like Genres, Year, Certificates etc.

**Comment by Omarscomin** [comment]
> I get neither of those 2 titles in the first 2 pages of Popular. I do, however, get Mr. Car on page one and My Fault on page two of a very popular addon that is in some ways similar to Fen. You wouldn't have screenshots would you?...
> 
> 
> 
> Anyway....
> 
> 
> 
> TRAKT: "Trending", "Trending Recent", "Top 10 Box Office" and "Most Watched" are all currently being offered in Fen. There really is not much more that trakt offers in the way of useful, pre-compiled lists to use.
> 
> 
> 
> IMDb: I currently offer 0 IMDb based pre-compiled lists. The main reason for this is that it takes an extra call to TMDb to get the metadata for each movie/show for these lists. Instead of 1 call, it's 2 as you can only get the IMDb ID from the lists, and TMDb requires a TMDb ID for accessing the metadata. So you need to first make a call to TMDb to get the TMDb ID from the IMDb ID. Plus IMDb doesn't offer a public API, so the lists are scraped from their website which is a lot slower. Since Fen is all about speed, this has always been an issue for me.
> 
> 
> 
> Also, Umbrella dev and myself have different tastes in lists. He has no problem with 2 lists both being called "Popular" for example, whilst I would not do that with Fen. There are likely some nice IMDb lists that would be useful to have though. A quick look shows "Most Voted", "Featured", "Most Popular" (with a rename).
> 
> 
> 
> I just wouldn't do 2 or more lists like Genres, Year, Certificates etc.

I'm away from my set up right now, but I'll get some screen shots when I can. 

Thank you for taking the time to explain all of that to me! I genuinely love gaining new perspectives :) I certainly understand why you'd be against implementing anything that could potentially slow Fen down. Also, I 100% agree with you about not wanting multiple lists with the same name. That would drive my OCD senses insane. With Umbrella, I only enable 1 of the "popular" list options, 1 of the "trending" list options, 1 of the "recently released" list options, and so on.

**Comment by kodifitzwell** [comment]
@Omarscomin looks like your list made it to "trending user lists."

![capture1](https://github.com/Tikipeter/repository.tikipeter/assets/105136042/c0295220-358e-422a-8f7c-0d556e09d30c)

**Comment by Omarscomin** [comment]
> @Omarscomin looks like your list made it to "trending user lists."
> 
> 
> 
> ![capture1](https://github.com/Tikipeter/repository.tikipeter/assets/105136042/c0295220-358e-422a-8f7c-0d556e09d30c)
> 
> 

Nice! This is the most famous I've ever been, and the most famous I'll ever be😂

**Comment by Omarscomin** [comment]
Just realized that I failed to provide an updated. My apologies!

I rebuilt databases and cleared cache, and those particular titles don't show up anymore. Not in the first several pages, at least. So, I guess my new primary reason for wanting those lists is the fact that I like that they display more titles per page :) I like the sorting better on IMDB, too. TMDB does have a bit of a tendency to display obscure foreign films in places that they don't quite belong. 

Anyhoo, thank you again for considering to cater to my nitpickyness :)
